### ISSUE TITLE
Junit5 Instrumented tests crash with JUnit 5.4.0

### ISSUE THREAD

**Issue by matejdro** [mannodermaus/android-junit5#150]
JUnit 5.4.0 seems to have broken instrumented tests that run on Android device/emulator.

Here are steps I did to reproduce this:

1. Create brand new emulator with `Android 9.0 (Google Play) x86_64` image
2. checkout this repo at https://github.com/mannodermaus/android-junit5/tree/7b9f490d9ba6e7eb6b91b99e6d2755c546354e4f
3. Start emulator
4. Run `./gradlew :sample:connectedExperimentalDebugAndroidTest` from the root of this repo
5. Tests will crash with following exception:

```
java.lang.ExceptionInInitializerError
at org.junit.platform.commons.util.StringUtils.isNotBlank(StringUtils.java:64)
at org.junit.platform.commons.util.Preconditions.notBlank(Preconditions.java:248)
at org.junit.platform.launcher.core.LauncherConfigurationParameters.<init>(LauncherConfigurationParameters.java:44)
at org.junit.platform.launcher.core.LauncherConfigurationParameters.<init>(LauncherConfigurationParameters.java:39)
at org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder.build(LauncherDiscoveryRequestBuilder.java:194)
at org.junit.platform.runner.JUnitPlatform.createDiscoveryRequest(JUnitPlatform.java:153)
at org.junit.platform.runner.JUnitPlatform.<init>(JUnitPlatform.java:124)
at org.junit.platform.runner.JUnitPlatform.<init>(JUnitPlatform.java:117)
at de.mannodermaus.junit5.AndroidJUnit5.<init>(Runner.kt:17)
at de.mannodermaus.junit5.RunnerKt.createJUnit5Runner(Runner.kt:22)
at de.mannodermaus.junit5.AndroidJUnit5Builder.runnerForClass(RunnerBuilder.kt:69)
at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
at androidx.test.internal.runner.AndroidRunnerBuilder.runnerForClass(AndroidRunnerBuilder.java:147)
at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
at androidx.test.internal.runner.TestLoader.doCreateRunner(TestLoader.java:73)
at androidx.test.internal.runner.TestLoader.getRunnersFor(TestLoader.java:104)
at androidx.test.internal.runner.TestRequestBuilder.build(TestRequestBuilder.java:789)
at androidx.test.runner.AndroidJUnitRunner.buildRequest(AndroidJUnitRunner.java:543)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:386)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
Caused by: java.lang.IllegalArgumentException: Unsupported flags: 256
at java.util.regex.Pattern.<init>(Pattern.java:1324)
at java.util.regex.Pattern.compile(Pattern.java:975)
at org.junit.platform.commons.util.StringUtils.<clinit>(StringUtils.java:36)
... 20 more
```

If you revert [5.4.0 update](https://github.com/mannodermaus/android-junit5/commit/4392c81252ee1f6679536e182de7028a2935375d), then this crash does not happen anymore.

Apparently JUnit is [now using UNICODE_CHARACTER_CLASS in its StringUtils](https://github.com/junit-team/junit5/commit/db8b3d4cdd588fad4660fbedaf1c6b2f68daed77) file, which is not supported on Android.

**Comment by matejdro** [comment]
Interestingly enough, https://github.com/mannodermaus/android-junit5/tree/c4717be89fd8a126766f31cb8336db3c0264a3c4 appears to work

**Comment by matejdro** [comment]
It appears to be an issue in JUnit itself. I have created parallel issue there: https://github.com/junit-team/junit5/issues/1800

**Comment by mannodermaus** [comment]
First time I've come across `UNICODE_CHARACTER_CLASS`. Let's wait for the 5.4.1 release then! I'll keep this issue open until that patch is out, and we have an update to the plugin as well.

**Comment by mannodermaus** [comment]
Released in `1.4.1.0` onwards! 👍
